src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunctionSet.cs

```diff
@@ -0,0 +1,344 @@
+
+        internal IEnumerable<TexlFunction> WithInvariantName(string name) => WithInvariantNameInternal(name);
+
+        private List<TexlFunction> WithInvariantNameInternal(string name) => _functionsInvariant.ContainsKey(name) ? _functionsInvariant[name] : new List<TexlFunction>();
+
+        internal IEnumerable<TexlFunction> WithInvariantName(string name, DPath ns) => _functionsInvariant.ContainsKey(name) ? new List<TexlFunction>(_functionsInvariant[name].Where(f => f.Namespace == ns)) : new List<TexlFunction>();
+
+        internal IEnumerable<TexlFunction> WithNamespace(DPath ns) => WithNamespaceInternal(ns);
+
+        private List<TexlFunction> WithNamespaceInternal(DPath ns) => _namespaces.ContainsKey(ns) ? _namespaces[ns] : new List<TexlFunction>();
+
+        internal IEnumerable<string> Enums => _enums;
+
+        internal TexlFunctionSet()
+        {
+            _functions = new Dictionary<string, List<TexlFunction>>();
+            _functionsInvariant = new Dictionary<string, List<TexlFunction>>();
+            _namespaces = new Dictionary<DPath, List<TexlFunction>>();
+            _enums = new List<string>();
+            _count = 0;
+        }
+
+        private TexlFunctionSet(Dictionary<string, List<TexlFunction>> functions, Dictionary<string, List<TexlFunction>> functionsInvariant, Dictionary<DPath, List<TexlFunction>> functionNamespaces, List<string> enums, int count)
+        {
+            _functions = functions ?? throw new ArgumentNullException(nameof(functions));
+            _functionsInvariant = functionsInvariant ?? throw new ArgumentNullException(nameof(functionsInvariant));
+            _namespaces = functionNamespaces ?? throw new ArgumentNullException(nameof(functionNamespaces));
+            _enums = enums ?? throw new ArgumentNullException(nameof(enums));
+            _count = count;
+        }
+
+        internal TexlFunctionSet(TexlFunction function)
+            : this()
+        {
+            if (function == null)
+            {
+                throw new ArgumentNullException(nameof(function));
+            }
+
+            _functions.Add(function.Name, new List<TexlFunction>() { function });
+            _functionsInvariant.Add(function.LocaleInvariantName, new List<TexlFunction>() { function });
+            _namespaces.Add(function.Namespace, new List<TexlFunction> { function });
+            _enums = new List<string>(function.GetRequiredEnumNames());
+            _count = 1;
+        }
+
+        internal TexlFunctionSet(IEnumerable<TexlFunction> functions)
+            : this()
+        {
+            if (functions == null)
+            {
+                throw new ArgumentNullException(nameof(functions));
+            }
+
+            foreach (var func in functions)
+            {
+                Add(func);
+            }
+        }
+
+        internal TexlFunctionSet(TexlFunctionSet other)
```

Comment:
>TexlFunctionSet [](http://example.com/codeflow?start=17&length=15)

Can this be private/ Only used by Clone()?

Reply:
Method has been removed